pkg/operator/nsfinalizercontroller/finalizer_controller.go

```diff
@@ -79,14 +84,22 @@ func (c finalizerController) sync() error {
 		return nil
 	}
 
-	pods, err := c.podLister.Pods(operatorclient.TargetNamespace).List(labels.Everything())
+	// allow one minute of grace for most things to terminate.
+	// TODO now that we have conditions, we may be able to check specific conditions
+	deletedMoreThanAMinute := ns.DeletionTimestamp.Time.Add(1 * time.Minute).Before(time.Now())
+	if !deletedMoreThanAMinute {
+		c.queue.AddAfter(c.namespaceName, 1*time.Minute)
+		return nil
+	}
+
+	pods, err := c.podLister.Pods(c.namespaceName).List(labels.Everything())
 	if err != nil {
 		return err
 	}
 	if len(pods) > 0 {
 		return nil
 	}
-	dses, err := c.dsLister.DaemonSets(operatorclient.TargetNamespace).List(labels.Everything())
+	dses, err := c.dsLister.DaemonSets(c.namespaceName).List(labels.Everything())
```

Comment:
do we expect more daemonsets in this namespace?

Reply:
> do we expect more daemonsets in this namespace?

we know we create them for apiservers today.